batteries/syntax/parser.luau

```diff
@@ -12,7 +12,18 @@ local function parseexpr(source: string): T.AstExpr
 	return luau.parseexpr(source)
 end
 
+export type ParseResult = {
+	root: T.AstStatBlock,
+	eof: T.Eof,
+}
+
+local function parsefile(source: string): ParseResult
+	local result = luau.parse(source)
+	return { root = result.root, eof = result.eof }
+end
+
 return {
 	parse = parse,
 	parseexpr = parseexpr,
+	parsefile = parsefile,
```

Comment:
Not 100% on this API name. It was unclear if this should've just replaced "parse", or if we still wanted the ability to parse a block independently of whether it was a whole file. Happy to change it if there's suggestions, or can do it in a followup when we standardise the whole API after it's complete.

Reply:
I think this is a fine name for now, i suppose

Reply:
cool, will leave it alone for now, and revisit it afterwards